### PR TITLE
Mass update: attack_types,detection_methods, tactics & techniques

### PR DESCRIPTION
**Pull request by morriscode**
duplicative tags were also stripped if they were an exact match for any of the new key values.

### CODE REVIEW

File: detection-rules/link_google_amp_suspicious_indicators.yml [hunk 1]
```diff
@@ -50,7 +50,19 @@ source: |
             or any(recipients.to, strings.icontains(..href_url.path, .email.email))
           )
   )
-tags: 
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Impersonation: Brand"
```
Comment (jkamdjou) [inline detection-rules/link_google_amp_suspicious_indicators.yml:56]:
```suggestion
```

File: detection-rules/link_google_amp_suspicious_indicators.yml [hunk 2]
```diff
@@ -50,7 +50,19 @@ source: |
...
   - "Open redirect"
-  - "Credential phishing"
+detection_methods:
+  - "Computer Vision"
+  - "Content analysis"
+  - "File analysis"
+  - "Header analysis"
```
Comment (jkamdjou) [inline detection-rules/link_google_amp_suspicious_indicators.yml:62]:
```suggestion
```

File: detection-rules/link_google_apps_script_macro.yml [hunk 3]
```diff
@@ -22,7 +22,12 @@ source: |
               and sender.email.domain.domain not in $sender_domains
           )
   )
-tags:
-  - "Suspicious link"
+attack_types:
+  - "Credential Phishing"
```
Comment (jkamdjou) [inline detection-rules/link_google_apps_script_macro.yml:26]: malware?

File: detection-rules/link_google_comment_script_macro.yml [hunk 4]
```diff
@@ -22,7 +22,13 @@ source: |
     // Or the Sender Display Name is not in your Org Display Names
     or not any($org_display_names, strings.istarts_with(sender.display_name, strings.concat(., " (Google ")))
   )
-tags:
-  - "Suspicious link"
-  - "Open redirect"  
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Open redirect"
```
Comment (jkamdjou) [inline detection-rules/link_google_comment_script_macro.yml:28]:
```suggestion
```

File: detection-rules/link_google_comment_script_macro.yml [hunk 5]
```diff
@@ -22,7 +22,13 @@ source: |
     // Or the Sender Display Name is not in your Org Display Names
     or not any($org_display_names, strings.istarts_with(sender.display_name, strings.concat(., " (Google ")))
   )
-tags:
-  - "Suspicious link"
-  - "Open redirect"  
+attack_types:
+  - "Credential Phishing"
```
Comment (jkamdjou) [inline detection-rules/link_google_comment_script_macro.yml:26]: malware?

File: detection-rules/link_html_smuggling_with_adobe_branding.yml [hunk 6]
```diff
@@ -36,9 +36,18 @@ source: |
...
-  - "Malware"
-  - "QakBot"
+  - "Malfam: QakBot"
+attack_types:
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Social engineering"
+detection_methods:
+  - "Archive analysis"
```
Comment (jkamdjou) [inline detection-rules/link_html_smuggling_with_adobe_branding.yml:46]:
```suggestion
```

File: detection-rules/link_ipfs_phishing.yml [hunk 7]
```diff
@@ -45,6 +44,14 @@ source: |
...
-tags:
-  - "Suspicious link"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Free file host"
+  - "Free subdomain host"
+  - "IPFS"
+detection_methods:
+  - "File analysis"
```
Comment (jkamdjou) [inline detection-rules/link_ipfs_phishing.yml:54]:
```suggestion
```

File: detection-rules/link_ipfs_phishing.yml [hunk 8]
```diff
@@ -45,6 +44,14 @@ source: |
             and sender.email.domain.domain not in $recipient_domains
         )
   )
-tags:
-  - "Suspicious link"
+attack_types:
+  - "Credential Phishing"
```
Comment (jkamdjou) [inline detection-rules/link_ipfs_phishing.yml:48]: malware?

File: detection-rules/link_login_or_captcha.yml [hunk 9]
```diff
@@ -40,8 +40,15 @@ source: |
...
-  - "Suspicious link"
-  - "Machine Learning"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Free subdomain host"
+  - "Social engineering"
+detection_methods:
+  - "Computer Vision"
+  - "Content analysis"
```
Comment (jkamdjou) [inline detection-rules/link_login_or_captcha.yml:50]:
```suggestion
```

File: detection-rules/link_microsoft_device_code_phish.yml [hunk 10]
```diff
@@ -41,6 +41,14 @@ source: |
...
-tags:
-  - "Suspicious link"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Social engineering"
+detection_methods:
+  - "Content analysis"
+  - "Header analysis"
```
Comment (jkamdjou) [inline detection-rules/link_microsoft_device_code_phish.yml:51]:
```suggestion
```

File: detection-rules/link_microsoft_impersonation_using_hosted_png.yml [hunk 11]
```diff
@@ -39,8 +39,13 @@ source: |
...
-  - "Suspicious link"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Social engineering"
+detection_methods:
+  - "Content analysis"
   - "Natural Language Understanding"
+  - "Sender analysis"
```
Comment (jkamdjou) [inline detection-rules/link_microsoft_impersonation_using_hosted_png.yml:50]:
```suggestion
  - "Sender analysis"
  - "URL analysis"
```

File: detection-rules/link_notion_file_share.yml [hunk 12]
```diff
@@ -24,7 +24,13 @@ source: |
...
             )
   )
-
-tags:
-  - "Suspicious link"
+attack_type:
+  - "Credential Phishing"
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/link_notion_file_share.yml:31]: free file host

File: detection-rules/link_outlook_left_to_right_exploit.yml [hunk 13]
```diff
@@ -9,7 +9,13 @@ source: |
...
   // this regex contains a Unicode U+200E character (may not be visible in many clients)
   and regex.contains(body.html.raw, 'base.{0,100}‎/>')
-tags: 
-  - "Suspicious link"
-  - "Office exploit"
+attack_type:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
+  - "Exploit"
```
Comment (jkamdjou) [inline detection-rules/link_outlook_left_to_right_exploit.yml:16]:
```suggestion
```

File: detection-rules/link_sharepoint_online_multiple_recipients_from_outside_org.yml [hunk 14]
```diff
@@ -11,9 +11,12 @@ source: |
   and length(recipients.to) > 40 // You may want to adjust this threshold if 40 is too aggressive
   and any(recipients.to, .email.domain.domain in $free_email_providers)
   and sender.display_name not in $org_display_names
-tags:
-  - "Suspicious link"
-  - "Suspicious content"
-  - "Suspicious subject"
-  - "Suspicious sender"
+attack_type:
+  - "Credential Phishing"
```
Comment (jkamdjou) [inline detection-rules/link_sharepoint_online_multiple_recipients_from_outside_org.yml:15]:
```suggestion
  - "Credential Phishing"
  - "Spam"
```

File: detection-rules/link_undisclosed_recipients_credphish.yml [hunk 15]
```diff
@@ -20,7 +20,14 @@ source: |
           )
       )
   )
-tags: 
-  - "Suspicious headers"
-  - "Suspicious link"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/link_undisclosed_recipients_credphish.yml:26]:
```suggestion
```

File: detection-rules/link_undisclosed_recipients_credphish.yml [hunk 16]
```diff
@@ -20,7 +20,14 @@ source: |
...
+detection_methods:
+  - "Computer Vision"
+  - "Content analysis"
```
Comment (jkamdjou) [inline detection-rules/link_undisclosed_recipients_credphish.yml:29]:
```suggestion
```

File: detection-rules/lookalike_sender_domain.yml [hunk 17]
```diff
@@ -24,7 +24,15 @@ source: |
...
+attack_type:
+  - "BEC/Fraud"
+  - "Credential Phishing"
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Impersonation: Brand"
   - "Lookalike domain"
+  - "Social engineering"
+detection_methods:
+  - "File analysis"
```
Comment (jkamdjou) [inline detection-rules/lookalike_sender_domain.yml:36]:
```suggestion
```

File: detection-rules/recipients_undisclosed_compauth_check.yml [hunk 18]
```diff
@@ -33,9 +33,13 @@ source: |
           )
       )
   )
-tags:
-  - "Suspicious sender"
-  - "Suspicious headers"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/recipients_undisclosed_compauth_check.yml:39]: does this belong

File: detection-rules/recipients_undisclosed_free_subdomain_host.yml [hunk 19]
```diff
@@ -17,7 +17,12 @@ source: |
   )
   and length(recipients.cc) == 0
   and length(recipients.bcc) == 0
-tags: 
-  - "Suspicious link"
-  - "Suspicious headers"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/recipients_undisclosed_free_subdomain_host.yml:23]: does this belong

File: detection-rules/sender_new_from_domain_first_time_sender.yml [hunk 20]
```diff
@@ -17,7 +17,8 @@ source: |
             )
   )
 tags:
-  - "Suspicious sender"
-  - "New domain"
   - "Attack surface reduction"
+detection_methods:
```
Comment (jkamdjou) [inline detection-rules/sender_new_from_domain_first_time_sender.yml:21]: missing attack_types

File: detection-rules/spam_campaign_excessive_display_text_with_keywords.yml [hunk 21]
```diff
@@ -24,7 +24,8 @@ source: |
               and sender.email.domain.domain not in $sender_domains
           )
       )
-severity: "low"
-tags:
+attack_types:
   - "Spam"
+detection_methods:
+  - "URL analysis"
```
Comment (jkamdjou) [inline detection-rules/spam_campaign_excessive_display_text_with_keywords.yml:30]:
```suggestion
  - "Content analysis"
```

File: detection-rules/spoof_dropbox.yml [hunk 22]
```diff
@@ -9,7 +9,14 @@ source: |
...
-  - "Brand impersonation"
-  - "Suspicious sender"
+attack_types:
+  - "Credential Phishing"
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Spoofing"
+detection_methods:
+  - "Content analysis"
```
Comment (jkamdjou) [inline detection-rules/spoof_dropbox.yml:19]:
```suggestion
```

File: detection-rules/spoofable_internal_domain_suspicious_signals.yml [hunk 23]
```diff
@@ -73,5 +73,18 @@ source: |
...
   )
 tags:
-  - "Spoof"
+  - "Attack surface reduction"
+attack_type:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Free file host"
+  - "Free subdomain host"
+  - "Impersonation: Brand"
```
Comment (jkamdjou) [inline detection-rules/spoofable_internal_domain_suspicious_signals.yml:82]:
```suggestion
```
is this brand?

File: detection-rules/stripe_invoice_abuse.yml [hunk 24]
```diff
@@ -23,7 +23,13 @@ source: |
...
   )
-tags: 
+attack_types:
+  - "BEC/Fraud"
   - "Callback Phishing"
-  - "Customer service fraud"
+tactics_and_techniques:
+  - "PDF"
+detection_methods:
+  - "Content analysis"
```
Comment (jkamdjou) [inline detection-rules/stripe_invoice_abuse.yml:32]:
```suggestion
```

File: detection-rules/suspicious_request_for_quote_html_smuggling.yml [hunk 25]
```diff
@@ -61,10 +61,16 @@ source: |
           )
       )
   )
-tags: 
-  - "RFP"
-  - "RFQ"
-  - "Suspicious attachment"
-  - "HTML smuggling"
+attack_types:
+  - "Credential Phishing"
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_html_smuggling.yml:65]: should this be BEC/Fraud?

File: detection-rules/suspicious_request_for_quote_html_smuggling.yml [hunk 26]
```diff
@@ -61,10 +61,16 @@ source: |
...
+tactics_and_techniques:
+  - "Evasion"
+  - "Free file host"
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_html_smuggling.yml:68]:
```suggestion
  - "HTML smuggling"
```

File: detection-rules/suspicious_request_for_quote_or_purchase.yml [hunk 27]
```diff
@@ -50,10 +50,13 @@ source: |
...
   )
-tags: 
-  - "Suspicious sender"
-  - "Fraud"
-  - "RFQ"
-  - "RFP"
+attack_types:
+  - "BEC/Fraud"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_or_purchase.yml:56]: freemail

File: detection-rules/suspicious_request_for_quote_or_purchase.yml [hunk 28]
```diff
@@ -50,10 +50,13 @@ source: |
...
+detection_methods:
+  - "Content analysis"
+  - "File analysis"
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_or_purchase.yml:59]:
```suggestion
```
Header analysis?

File: detection-rules/suspicious_sharepoint_file_share_undisclosed_recipients.yml [hunk 29]
```diff
@@ -6,13 +6,19 @@ type: "rule"
...
   and strings.icontains(subject.subject, "shared")
   and any(body.links, .href_url.domain.root_domain == "sharepoint.com")
-tags: 
-  - "Suspicious headers"
-  - "Suspicious Link"
+attack_types:
+  - "Credential Phishing"
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/suspicious_sharepoint_file_share_undisclosed_recipients.yml:19]: free subdomain

File: detection-rules/suspicious_shipping_notification.yml [hunk 30]
```diff
@@ -7,20 +7,25 @@ source: |
...
-  
+
   // email is not personalized with recipients name
   and any(recipients.to, any(ml.nlu_classifier(coalesce(body.html.display_text, body.plain.raw)).entities, .text == ..email.local_part))
-tags: 
+attack_types:
+  - "Credential Phishing"
+  - "Spam"
+tactics_and_techniques:
+  - "Evasion"
```
Comment (jkamdjou) [inline detection-rules/suspicious_shipping_notification.yml:27]: does this belong

File: detection-rules/uri_handler_search_ms.yml [hunk 31]
```diff
@@ -17,7 +17,12 @@ source: |
...
-tags:
-  - "Suspicious attachment"
-  - "Malware"
-id: "ee27d9c0-2d7e-5f6d-85be-73f3bf5b7cb1"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
+detection_methods:
+  - "Content analysis"
```
Comment (jkamdjou) [inline detection-rules/uri_handler_search_ms.yml:25]:
```suggestion
```

File: detection-rules/uri_handler_search_ms.yml [hunk 32]
```diff
@@ -17,7 +17,12 @@ source: |
               )
           )
   )
-tags:
-  - "Suspicious attachment"
-  - "Malware"
-id: "ee27d9c0-2d7e-5f6d-85be-73f3bf5b7cb1"
+attack_types:
+  - "Credential Phishing"
```
Comment (jkamdjou) [inline detection-rules/uri_handler_search_ms.yml:21]: i think this is malware

File: detection-rules/urlhaus_malicious_dom_in_body_or_pdf.yml [hunk 33]
```diff
@@ -45,8 +45,16 @@ source: |
...
-  - "Suspicious attachments"
-  - "URLhaus"
-  - "Abusech"
+  - "Abusech: URLhaus"
+attack_types:
+  - "Credential Phishing"
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Free file host"
+  - "Free subdomain host"
```
Comment (jkamdjou) [inline detection-rules/urlhaus_malicious_dom_in_body_or_pdf.yml:54]: nots

File: detection-rules/attachment_adobe_image_lure_fts.yml [hunk 34]
```diff
@@ -25,8 +24,14 @@ source: |
...
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Image as content"
+  - "Impersonation: Brand"
+detection_methods:
+  - "Content analysis"
   - "Computer Vision"
+  - "Sender analysis"
+  - "URL analysis"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_adobe_image_lure_fts.yml:36]: OCR isn't being called explicitly in the rule, 
Logo detect is covered via computer vision, and the string check via content analysis. But you and i both know the strings array could be being populated in part from OCR, but I found it confusing to label it without it being present in the rule.

File: detection-rules/impersonation_apple.yml [hunk 35]
```diff
@@ -13,7 +13,12 @@ source: |
...
   and sender.email.domain.root_domain !~ 'apple.com'
   and sender.email.email not in $recipient_emails
-tags:
-  - "Brand impersonation"
-  - "Suspicious sender"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Social engineering"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/impersonation_apple.yml:20]: This is Levenshtein on the display name not the domain.

File: detection-rules/impersonation_dropbox.yml [hunk 36]
```diff
@@ -17,7 +17,14 @@ source: |
...
   )
   and sender.email.email not in $recipient_emails
-tags: 
-  - "Brand impersonation"
-  - "Suspicious sender"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Social engineering"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/impersonation_dropbox.yml:24]: levenshtein on display name not domain.

File: detection-rules/link_google_amp_suspicious_indicators.yml (hunk 1 above)
Comment (morriscode) [reply detection-rules/link_google_amp_suspicious_indicators.yml:56]: There's a brand logo check in here.

File: detection-rules/lookalike_sender_domain.yml [hunk 37]
```diff
@@ -24,7 +24,15 @@ source: |
...
       )
   )
-tags:
-  - "Suspicious sender"
+attack_type:
+  - "BEC/Fraud"
+  - "Credential Phishing"
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Impersonation: Brand"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/lookalike_sender_domain.yml:32]: *nod*

File: detection-rules/spoofable_internal_domain_suspicious_signals.yml (hunk 23 above)
Comment (morriscode) [reply detection-rules/spoofable_internal_domain_suspicious_signals.yml:82]: It's a org spoof so I believe so?

File: detection-rules/attachment_html_smuggling_eval_atob.yml [hunk 38]
```diff
@@ -16,7 +16,16 @@ source: |
...
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
   - "HTML smuggling"
+  - "Scripting"
+detection_methods:
+  - "Archive analysis"
+  - "Content analysis"
+  - "File analysis"
+  - "HTML analysis"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_html_smuggling_eval_atob.yml:30]: scan.javascript is not being called, but rather a generic strings check for javascript chars. I can go either way here, but programatically being able to make this decision in the future requires an anchor like "scan.javascript" being able to infer the intent of string usage feels unsustainable or manual at the very least.

File: detection-rules/attachment_html_smuggling_hex_strings.yml [hunk 39]
```diff
@@ -23,7 +23,14 @@ source: |
...
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Evasion"
   - "HTML smuggling"
+detection_methods:
+  - "Archive analysis"
+  - "Content analysis"
+  - "File analysis"
+  - "HTML analysis"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_html_smuggling_hex_strings.yml:35]: scan.javascript is not being called, but rather a generic strings check for javascript chars. I can go either way here, but programatically being able to make this decision in the future requires an anchor like "scan.javascript" being able to infer the intent of string usage feels unsustainable or manual at the very least.

File: detection-rules/attachment_macro_dll_loader.yml [hunk 40]
```diff
@@ -34,8 +34,15 @@ source: |
...
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Exploit"
+  - "LNK"
   - "Macros"
+  - "Scripting"
+detection_methods:
+  - "Archive analysis"
+  - "File analysis"
+  - "YARA"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_macro_dll_loader.yml:47]: macro classifier is not being called.

File: detection-rules/attachment_mht_embedded_vbscript.yml [hunk 41]
```diff
@@ -33,7 +33,15 @@ source: |
...
           )
   )
-tags:
-  - "Suspicious attachment"
-  - "Malware"
+attack_types:
+  - "Malware/Ransomware"
+tactics_and_techniques:
+  - "Evasion"
+  - "HTML smuggling"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_mht_embedded_vbscript.yml:40]: An MHT file is a webpage archive, it's essentially a MIME HTML type no?

File: detection-rules/attachment_rdp_connection_file.yml [hunk 42]
```diff
@@ -20,6 +20,10 @@ source: |
           )
       )
   )
-tags:
-  - "Suspicious attachment"
+attack_types:
+  - "Malware/Ransomware"
+  - "Credential Phishing"
```
[the remark this answers is not in the document]
Comment (aidenmitchell) [reply detection-rules/attachment_rdp_connection_file.yml:25]: To add, authenticating to an RDP server can mean sending it your NTLM password hash, which is a pretty easy way to steal someone's password.
https://syfuhs.net/how-authentication-works-when-you-use-remote-desktop

File: detection-rules/impersonation_amazon.yml [hunk 43]
```diff
@@ -44,7 +44,12 @@ source: |
...
   and sender.email.email not in $recipient_emails
   and sender.email.domain.domain not in $org_domains
-tags:
-  - "Brand impersonation"
-  - "Suspicious sender"
+attack_types:
+  - "Credential Phishing"
+tactics_and_techniques:
+  - "Impersonation: Brand"
+  - "Social engineering"
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/impersonation_amazon.yml:51]: Not levenshtein on domain name, just display name.

File: detection-rules/link_outlook_left_to_right_exploit.yml (hunk 13 above)
Comment (morriscode) [reply detection-rules/link_outlook_left_to_right_exploit.yml:16]: description: |
  The exploit involves tricking Outlook for Windows into displaying a fake domain while opening another one.

File: detection-rules/link_undisclosed_recipients_credphish.yml (hunk 15 above)
Comment (morriscode) [reply detection-rules/link_undisclosed_recipients_credphish.yml:26]: I personally view the technique of masking recipients as evasion
      // No Recipients
      length(recipients.to) == 0
      or all(recipients.to, .display_name == "Undisclosed recipients")

File: detection-rules/sender_new_from_domain_first_time_sender.yml (hunk 20 above)
Comment (morriscode) [reply detection-rules/sender_new_from_domain_first_time_sender.yml:21]: intentional

**Review by jkamdjou — APPROVED** [review]

**Pull request merged**